src/Http/Http/src/BindingAddress.cs

```diff
@@ -203,10 +203,11 @@ public static BindingAddress Parse(string address)
         var port = 0;
 
         var hasSpecifiedPort = false;
-        if (!isUnixPipe)
+        if (!isUnixPipe && !isNamedPipe)
         {
+            // Verify not a loopback uri.
             var portDelimiterStart = address.LastIndexOf(':', pathDelimiterStart - 1, pathDelimiterStart - schemeDelimiterEnd);
-            if (portDelimiterStart >= 0)
+            if (portDelimiterStart >= 0 && address.Substring(pathDelimiterStart - 1, 1) != "]")
```

Comment:
You could index into the address and compare against `]` to avoid the allocation.

`pathDelimiterStart - 1` is calculated already above.